atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionLockWatchingCacheView.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableMap;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.GuardedValue;
+import com.palantir.atlasdb.keyvalue.api.ImmutableGuardedValue;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.lock.v2.LockWatch;
+
+public class TransactionLockWatchingCacheView {
+    public static final TransactionLockWatchingCacheView EMPTY = NoOpLockWatchingCache.INSTANCE
+            .getView(0, ImmutableMap.of(), null);
+
+    private final long startTimestamp;
+    private final BiFunction<TableReference, Cell, LockWatch> cellToWatch;
```

Comment:
Should this be an interface? It looks like we might use it frequently in the future?